tactics/proof_global.ml

```diff
@@ -18,9 +18,9 @@ module NamedDecl = Context.Named.Declaration
 
 type proof_object =
   { name : Names.Id.t
+  (* [name] only used in the STM *)
   ; entries : Evd.side_effects Declare.proof_entry list
   ; uctx: UState.t
```

Comment:
Note that after this the proof object is basically a `uctx` and the `entr{y,ies}` , that is one of the goals here, and puts them in the same foot that what `DeclareDef.prepare` returns. In fact, if not for the name we could unify with `ClosedDef.t` already.